File: python/semantic_kernel/data/filters/equal_to_filter_clause.py
# Copyright (c) Microsoft. All rights reserved.

from pydantic import Field

from semantic_kernel.data.filters.filter_clause_base import FilterClauseBase
from semantic_kernel.kernel_pydantic import KernelBaseModel


class EqualTo(FilterClauseBase, KernelBaseModel):
    """A filter clause for an equals comparison."""

    filter_clause_type: str = Field("equal_to", init=False)  # type: ignore

    field_name: str
    value: str

File: dotnet/src/Connectors/VectorData.Abstractions/FilterClauses/EqualToFilterClause.cs
// Copyright (c) Microsoft. All rights reserved.

namespace Microsoft.Extensions.VectorData;

/// <summary>
/// Represents a filter clause that filters using equality of a field value.
/// </summary>
public sealed class EqualToFilterClause : FilterClause
{
    /// <summary>
    /// Initializes a new instance of the <see cref="EqualToFilterClause"/> class.
    /// </summary>
    /// <param name="fieldName">Field name.</param>
    /// <param name="value">Field value.</param>
    public EqualToFilterClause(string fieldName, object value)
    {
        this.FieldName = fieldName;
        this.Value = value;
    }

    /// <summary>
    /// Gets the field name to match.
    /// </summary>
    public string FieldName { get; private set; }

    /// <summary>
    /// Gets the field value to match.
    /// </summary>
    public object Value { get; private set; }
}
